lib/msal-core/test/ScopeSet.spec.ts

```diff
@@ -1,16 +1,205 @@
+        });
+
+        it("should return false if cachedScopes and scopes are both empty arrays", () => {
+            const cachedScopes = [];
+            const scopes = [];
+            expect(ScopeSet.isIntersectingScopes(cachedScopes, scopes)).to.eql(false);
+        });
+    });
+
+    describe("trimAndConvertToLowerCase", function() {
+        it("should downcase and remove all whitespace from all scope strings in scopes array passed in", () => {
+            const scopeSet = ["S1", " S2", " S3 "];
+            expect(ScopeSet.trimAndConvertArrayToLowerCase(scopeSet)).to.be.deep.eq(["s1", "s2", "s3"]);         
+        });
+    });
+
+    describe("trimAndConvertToLowerCase", () => {
+        it("should downcase and remove all whitespace from the scope passed in", () => {
+            const scope = " S1 ";
+            expect(ScopeSet.trimAndConvertToLowerCase(scope)).to.be.eq("s1");
+        });
+    });
+
+    describe("removeElement", function() {
+        it("should return a filtered scopes array with the scope passed in removed", function() {
+            const unfilteredScopes = ["s1", "s2"];
+            const scopeToRemove = "s1";
+            const filteredScopes = ScopeSet.removeElement(unfilteredScopes, scopeToRemove);
+            expect(filteredScopes).to.not.eq(unfilteredScopes);
+            expect(filteredScopes).to.not.include(scopeToRemove);
+        });
+    });
+
+    describe("validateInputScope", function() {
+        it("should not throw any errors when a single scope is passed in scopes array", () => {
+            const scopes = ["s1"];
+            expect(() => ScopeSet.validateInputScope(scopes)).to.not.throw();
+        });
+
+        it("should not throw any errors when multiple scopes are passed in scopes array", () => {
+            const scopes = ["s1", "s2"];
+            expect(() => ScopeSet.validateInputScope(scopes)).to.not.throw();
+        });
+
+        it("should throw createScopesRequiredError if scopes are empty or null", function() {
+            const scopes = null;
+            let clientConfigError;
 
-describe("ScopeSet.ts Unit Tests", () => {
+            try {
+                ScopeSet.validateInputScope(scopes);
+            } catch (e) {
+                clientConfigError = e;
+            }
 
-    it("tests trimAndConvertArrayToLowerCase", () => {
-        const scopeSet = ["S1", " S2", " S3 "];
-        expect(ScopeSet.trimAndConvertArrayToLowerCase(scopeSet)).to.be.deep.eq(["s1", "s2", "s3"]);         
+            expect(clientConfigError instanceof ClientConfigurationError).to.eq(true);
+            expect(clientConfigError.errorCode).to.equal(ClientConfigurationErrorMessage.scopesRequired.code);
+            expect(clientConfigError.message).to.contain(ClientConfigurationErrorMessage.scopesRequired.desc);
+        });
+        
+        it("should throw createScopesNonArrayError if scopes is not an array object", function() {
+            const scopes = {};
```

Comment:
Can we make this a string of space delimited example scopes as that is a more likely scenario?

Reply:
Done.